### PR TITLE
Make releaseable

### PR DESCRIPTION
**Pull request by keeslinp**
The daily settings updater caught up to me and I can't push to master anymore 😁 . 

terraform registry recommends using goreleaser so this PR adds that and the github action that they recommend. The only things that are needed after this is adding the secrets and getting the terraform registry approved access to lifeomic.

Also added some docs so that something will appear on the registry page.

Basically everything in this PR is pulled from https://www.terraform.io/registry/providers/publishing

### CODE REVIEW

**Review by ynotdraw — DISMISSED** [review]
Not a go pro by any means, but this looks good to me other than the small typo in `docs/index.md` 👍

**Comment by keeslinp** [comment]
> Not a go pro by any means, but this looks good to me other than the small typo in `docs/index.md` 👍

Fixed 👍 . I also added a README and fixed a secret name (now that we actually have the secret on the project).

I am definitely not a go expert either haha. Even just since making the second one I already have lots of things I want to fix in this one, but I don't want to get too hung up yet. I'll probably do another pass once everything is working to clean up code.

**Review by ynotdraw — APPROVED** [review]
🎉
